Code/Framework/AzToolsFramework/AzToolsFramework/Metadata/UuidUtils.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/Metadata/UuidUtils.h>
+#include <AzToolsFramework/Metadata/MetadataManager.h>
+#include <AzToolsFramework/Metadata/UuidEntry.h>
+
+namespace AzToolsFramework
+{
+    void UuidUtilComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<UuidUtilComponent, AZ::Component>();
+        }
+    }
+
+    bool UuidUtilComponent::CreateSourceUuid(AZ::IO::PathView absoluteFilePath, AZ::Uuid uuid)
+    {
+        auto* metadataInterface = AZ::Interface<IMetadataRequests>::Get();
+
+        if (!metadataInterface)
+        {
+            AZ_Assert(metadataInterface, "Programmer Error - IMetadataRequests interface is not available");
+            return false;
+        }
+
+        AzToolsFramework::UuidEntry entry;
+
+        if (metadataInterface->GetValue(absoluteFilePath, UuidKey, entry) && !entry.m_uuid.IsNull())
+        {
+            return false;
```

Comment:
Should there be a warning, error, or log here, or is it that handled in the call to GetValue? The site calling CreateSourceUuid probably can't handle it because it won't know if the false return is due to GetValue failing, the metadataInterface not being available, or SetValue failing.

Reply:
I have another PR coming up which will overhaul the error handling